ariadne/cardano/src/Ariadne/Wallet/Backend.hs

```diff
@@ -41,12 +40,16 @@ createWalletBackend walletConfig sendWalletEvent = do
         (openLocalStateFrom walletAcidDbPathPlaceholder defDB)
         closeAcidState
 
-    (us :: TVar UserSecret, addUs :: TVar UserSecret -> IO ()) <- newInitFuture "UserSecret"
-    PassiveWalletLayer{..} <- passiveWalletLayerComponent
-        (usingLoggerName "passive-wallet" ... logMessage) us acidDb
-    let refresh = refreshState acidDb walletSelRef sendWalletEvent
-        applyHook = const refresh <=< _pwlApplyBlocks
-        rollbackHook = const refresh <=< _pwlRollbackBlocks
+    -- TODO: configurable path to keyfile
+    keystore <- keystoreComponent RemoveKeystoreIfEmpty "secret-mainnet.key"
+    pwl <- passiveWalletLayerWithDBComponent
```

Comment:
Maybe let's not use `acidDB` outside wallet layer at all? And then create it inside `PassiveWalletLayer` creation function and not expose outside? It should be trivial currently, because it's used only in two places here and `_pwlGetDBSnapshot` is enough.
It will highlight that all code which deals with `acid-state` is hidden behind `PassiveWalletLayer`.